src/ConfigProvider.php

```diff
@@ -0,0 +1,132 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Laminas\Mvc;
+
+use Psr\Container\ContainerInterface;
+use Laminas\EventManager\EventManager;
+use Laminas\EventManager\EventManagerAwareInterface;
+use Laminas\EventManager\EventManagerInterface;
+use Laminas\EventManager\SharedEventManager;
+use Laminas\EventManager\SharedEventManagerInterface;
+use Laminas\Mvc\Controller\ControllerManager;
+use Laminas\Mvc\Service\ApplicationFactory;
+use Laminas\Mvc\Service\EventManagerFactory;
+use Laminas\Mvc\Service\HttpDefaultRenderingStrategyFactory;
+use Laminas\Mvc\Service\HttpExceptionStrategyFactory;
+use Laminas\Mvc\Service\HttpRouteNotFoundStrategyFactory;
+use Laminas\Mvc\Service\SendResponseListenerFactory;
+use Laminas\Mvc\Service\ViewFactory;
+use Laminas\Mvc\Service\ViewPhpRendererFactory;
+use Laminas\Mvc\Service\ViewPhpRendererStrategyFactory;
+use Laminas\ServiceManager\Factory\InvokableFactory;
+
+class ConfigProvider
+{
+    public function __invoke(): array
+    {
+        return [
+            'dependencies' => $this->getDependencies(),
+        ];
+    }
+
+    public function getDependencies(): array
+    {
+        return [
+            'aliases' => [
+                'EventManagerInterface' => EventManager::class,
+                EventManagerInterface::class => 'EventManager',
+                SharedEventManager::class => 'SharedEventManager',
+                'SharedEventManagerInterface' => 'SharedEventManager',
+                SharedEventManagerInterface::class => 'SharedEventManager',
+                'application' => 'Application',
+                'Config' => 'config',
+                'configuration' => 'config',
+                'Configuration' => 'config',
+                'HttpDefaultRenderingStrategy' => View\Http\DefaultRenderingStrategy::class,
+                'request' => 'Request',
+                'response' => 'Response',
```

Comment:
They definitely do not  belong to services.
I plan to drop them not only from container but also from bootstrap event. New event would need to be introduced to do request aware bootstap before routing. I think I did that work in zend-mvc but abandoned when we migrated to laminas.

We discussed it long time ago, and it should be on the old rfc: it would make sense to inject them to the run method with laminas-httphandlerrunner when mvc is  migrated to PSR and something similar before that.